contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,229 @@
  * SPDX-License-Identifier: UNLICENSED
  */
 pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import {Initializable} from "./packages/oz/upgradeability/Initializable.sol";
+import {SafeMath} from "./packages/oz/SafeMath.sol";
+import {OtokenInterface} from "./interfaces/OtokenInterface.sol";
+import {OracleInterface} from "./interfaces/OracleInterface.sol";
+import {AddressBookInterface} from "./interfaces/AddressBookInterface.sol";
+import {FixedPointInt256} from "./libs/FixedPointInt256.sol";
+
+// TODO: use this from libs/MarginAccout
+struct Vault {
+    uint256[] shortAmounts;
+    uint256[] longAmounts;
+    uint256[] collateralAmounts;
+    address[] shortOtokens;
+    address[] longOtokens;
+    address[] collateralAssets;
+}
 
 /**
  *
  */
-// solhint-disable-next-line no-empty-blocks
-contract MarginCalculator {
+contract MarginCalculator is Initializable {
+    using SafeMath for uint256;
+    using FixedPointInt256 for int256;
+    address public addressBook;
+
+    function init(address _addressBook) external initializer {
+        addressBook = _addressBook;
+    }
+
+    /**
+     * @notice Return the cash value of an expired oToken.
+     * @dev For call return = Max (0, ETH Price - oToken.strike)
+     * @dev For put return Max(0, oToken.strike - ETH Price)
+     * @param _otoken otoken address
+     */
+    function getExpiredCashValue(address _otoken) public view returns (uint256) {
+        require(_otoken != address(0), "MarginCalculator: Can't calculate cash value for non-otokens.");
+
+        OtokenInterface otoken = OtokenInterface(_otoken);
+        uint256 strikePrice = otoken.strikePrice();
```

Comment:
check that you can't get cash value for non-expired options

Reply:
That will revert in L48, do we need another check for expiry?
```
require(isFinalized, "MarginCalculator: Oracle price not finalized yet.");
```

Reply:
Its better to have that check as well in case there is some issue in oracle where isFinalized & expiry break

Reply:
Okay, just added!